android/test-app-native-modules.gradle

```diff
@@ -0,0 +1,282 @@
+
+{{ packageImports }}
+
+public class PackageList {
+  private Application application;
+  private ReactNativeHost reactNativeHost;
+  private MainPackageConfig mConfig;
+
+  public PackageList(ReactNativeHost reactNativeHost) {
+    this(reactNativeHost, null);
+  }
+
+  public PackageList(Application application) {
+    this(application, null);
+  }
+
+  public PackageList(ReactNativeHost reactNativeHost, MainPackageConfig config) {
+    this.reactNativeHost = reactNativeHost;
+    mConfig = config;
+  }
+
+  public PackageList(Application application, MainPackageConfig config) {
+    this.reactNativeHost = null;
+    this.application = application;
+    mConfig = config;
+  }
+
+  private ReactNativeHost getReactNativeHost() {
+    return this.reactNativeHost;
+  }
+
+  private Resources getResources() {
+    return this.getApplication().getResources();
+  }
+
+  private Application getApplication() {
+    if (this.reactNativeHost == null) return this.application;
+    return this.reactNativeHost.getApplication();
+  }
+
+  private Context getApplicationContext() {
+    return this.getApplication().getApplicationContext();
+  }
+
+  public ArrayList<ReactPackage> getPackages() {
+    return new ArrayList<>(Arrays.<ReactPackage>asList(
+      new MainReactPackage(mConfig){{ packageClassInstances }}
+    ));
+  }
+}
+"""
+
+class ReactNativeModules {
+    private Logger logger
+    private String jsAppDir
+    private ArrayList<HashMap<String, String>> reactNativeModules
+
+    private static String LOG_PREFIX = ":ReactNative:"
+    private static String REACT_NATIVE_CLI_BIN = "node_modules${File.separator}" +
+            "@react-native-community${File.separator}" +
+            "cli${File.separator}" +
+            "build${File.separator}" +
+            "index.js"
+
+    ReactNativeModules(Logger logger, String jsAppDir) {
+        this.logger = logger
+        this.jsAppDir = jsAppDir
+        this.reactNativeModules = this.getReactNativeConfig()
+    }
+
+    /**
+     * Include the react native modules android projects and specify their project directory
+     */
+    void addReactNativeModuleProjects(DefaultSettings defaultSettings) {
+        reactNativeModules.forEach { reactNativeModule ->
+
+            String nameCleansed = reactNativeModule["nameCleansed"]
+            String androidSourceDir = reactNativeModule["androidSourceDir"]
+
+            defaultSettings.include(":${nameCleansed}")
+            defaultSettings.project(":${nameCleansed}").projectDir = new File("${androidSourceDir}")
+        }
+    }
+
+    /**
+     * Adds the react native modules as dependencies to the users `app` project
+     */
+    void addReactNativeModuleDependencies(Project appProject) {
+        reactNativeModules.forEach { reactNativeModule ->
+            def nameCleansed = reactNativeModule["nameCleansed"]
+            appProject.dependencies {
+                // TODO(salakar): are other dependency scope methods such as `api` required?
```

Comment:
Was this copied from somewhere? What was its license? Do we need to put that copyright notice here?

Reply:
That's correct. The whole file is a fork of the auto-linking plugin from the cli tools. I will add the header.